### PR TITLE
copr: support sum

### PR DESCRIPTION
**Pull request by BusyJay**
@ngaut @coocood @shenli @qiuyesuifeng PTAL

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

**Comment by BusyJay** [comment]
ping

File: src/server/coprocessor/aggregate.rs [hunk 1]
```diff
@@ -59,3 +61,36 @@ impl AggrFunc for First {
...
+    fn update(&mut self, mut args: Vec<Datum>) -> Result<()> {
+        if args.len() != 1 {
+            return Err(box_err!("sum only support one column, but got {}", args.len()));
+        }
+        let a = args.pop().unwrap();
+        if a == Datum::Null {
+            return Ok(());
+        }
+        let res = match self.res.take() {
+            Some(b) => box_try!(evaluator::eval_arith(a, b, Datum::checked_add)),
```
Comment (shenli) [inline src/server/coprocessor/aggregate.rs]: When meet the first not null row, b will be null. Will eval_arith return null for a+b?

Comment (BusyJay) [inline src/server/coprocessor/aggregate.rs]: When meet the first not null row, `self.res.take()` will return `None`.

File: tests/coprocessor/test_select.rs [hunk 2]
```diff
@@ -339,6 +346,36 @@ fn test_aggr_first() {
...
+    let (mut store, mut end_point, ti) = initial_data(count);
+
+    let mut col = Expr::new();
+    col.set_tp(ExprType::ColumnRef);
+    col.mut_val().encode_i64(1).unwrap();
+    let mut expr = Expr::new();
+    expr.set_tp(ExprType::Sum);
+    expr.mut_children().push(col);
+
+    let req = prepare_sel(&mut store, &ti, None, None, vec![expr], vec![3]);
```
Comment (shenli) [inline tests/coprocessor/test_select.rs]: So the req is "select sum(first_column) from t"?

Comment (BusyJay) [inline tests/coprocessor/test_select.rs]: `select sum(first_col) from t group by second_col`.

**Comment by shenli** [comment]
LGTM

**Pull request merged**